fix : post problème | update probleme

SamEPK opened the pull request:
ajout date création post

SamEPK committed:
fix : post problème | update probleme

ajout date création post

SamEPK committed:
post commentaire

julienEcole commented on typescript-mysql-quickstart/source/controllers/vecoleo/probleme/commentaire.ts:
```diff
@@ -20,15 +20,22 @@ const NAMESPACE = 'commentaire';
 const createCommentaire = async (req: Request, res: Response, next: NextFunction) => {  //TODO
     logging.info(NAMESPACE, 'Inserting commentaire');
-
+    console.log("mon body = ", req.body)
```
a enlever lors du rendu final

julienEcole commented on typescript-mysql-quickstart/source/controllers/vecoleo/probleme/probleme.ts:
```diff
@@ -17,93 +17,92 @@ const NAMESPACE = 'probleme';
     }
-    const adresse : string = req.body.adresse;
-    const titre : string = req.body.titre;
+    const adresse: number = req.body.adresse;
```
si tu modifie le type il faut le parseInt (sinon il sera traité comme un string dans la var donc = NaN) et dans la query il faut enlever les " qui entourent la valeur sinon ce seras interpréter comme un string donc gén-èreras une erreur

julienEcole commented on typescript-mysql-quickstart/source/controllers/utilisateur.ts:
```diff
@@ -106,21 +106,21 @@ const updateOneUserById = async (req: Request, res: Response, next: NextFunction
-        res.send("le mot de passe n'est pas assez long (8 caractère minimum dont un minuscule, une majuscule & un caractère spécial)");
-        return;
-    }
+    // if(!isNumber.test(telephone)){
```
quand a ces tests il suffirait de remplacer if(!isNumber.test(telephone)) par if(telephone && !isNumber.test(telephone))

julienEcole replied:
et c'est la même pour les autres tests afin de ne pas tenter la regex sur un UNDEFINED

julienEcole committed:
feat : début gestion image

julienEcole committed:
Merge branch 'angular/fix' of https://github.com/julienEcole/projetAnnuel2023 into angular/fix

julienEcole committed:
fix : correction du fichier sql pour rennommer certains champ de commentaire & probleme
fix : la regex du telephone était toujours testé dans utilisateur même si aucun numero n'était envoyé
fix : modifier le titre (maintenant = resume) uniquement d'un commentaire ne cause plus une erreur de syntaxe sql
fix : rennomage complet de titre en objet pour la table probleme & son controlleur
fix : certains commentaire //DEBUG ont été ajouté pour spécifier les lignes a enlever lors du rendu final

SamEPK committed:
post probleme fix + commentaire fix

on peut désormais poster + d'1 commentaire par personne

SamEPK committed:
BackOffice : Commentaire | Page 404

julienEcole committed:
fix : petite erreur dans atelier,
feat : activation des routes de la table image

SamEPK committed:
Get Atelier + Page Atelier Unique

SamEPK committed:
Merge branch 'angular/fix' of https://github.com/julienEcole/projetAnnuel2023 into angular/fix

SamEPK committed:
fix regex

julienEcole committed:
petit fix sur commentaire & image

julienEcole committed:
Merge branch 'angular/fix' of https://github.com/julienEcole/projetAnnuel2023 into angular/fix

julienEcole committed:
fix : activation des routes de probleme_image,
feat : ajout des routes pour la suppression de toutes les liaisons via l'id d'une image ou l'id d'un probleme

SamEPK committed:
Page Profil unique | Fix Update profil | Fix Regex

SamEPK committed:
Restriction Page Profil

+ Redirection Profil sur pseudo des post

julienEcole committed:
feat : ajout des routes pour role_utilisateurs

julienEcole committed:
Merge branch 'angular/fix' of https://github.com/julienEcole/projetAnnuel2023 into angular/fix

julienEcole committed:
fix : petit erreur sql sur la methode put corrigé

julienEcole committed:
fix : de nombreux bug de corriger sur la table de liason probleme_image et ses routes qui sont maitenant vérifié

julienEcole committed:
feat : un utilisateur peut maintenant être lié a une image sur la table utilisateur pour stocker un avatar (image visible lors de l'envoie de certains commentaire
fix : certaines methode patch avaient un bug si on ne mettait a jour qu'un seul argument sur les fichiers mis a jour

julienEcole committed:
factorize : dossier API viré

julienEcole committed:
feat : ajout d'une image associé a un utilisateur
